finetuner/models.py

```diff
@@ -0,0 +1,158 @@
+import abc
+from typing import Any, Dict, List, Optional, Tuple, TypeVar
+
+ModelStubType = TypeVar('ModelStubType', bound='_ModelStub')
+
+
+class _ModelStub(metaclass=abc.ABCMeta):
+    name: str
+    description: str
+    task: str
+    output_dim: Optional[int]
+    architecture: str
+    options: Dict[str, Any]
+
+    def __init__(self, **kwargs):
+        self.options = kwargs
+
+    @staticmethod
+    def header() -> Tuple[str, ...]:
+        """Get table header."""
+        return 'name', 'task', 'output_dim', 'architecture', 'description'
+
+    @classmethod
+    def row(cls) -> Tuple[str, ...]:
+        """Get table row."""
+        return (
+            cls.name,
+            cls.task,
+            str(cls.output_dim),
+            cls.architecture,
+            cls.description,
+        )
+
+
+class MLP(_ModelStub):
+    """MLP model stub.
+
+    :param input_size: Size of the input representations.
+    :param hidden_sizes: A list of sizes of the hidden layers. The last hidden size is
+        the output size.
+    :param bias: Whether to add bias to each layer.
+    :param activation: A string to configure activation function, `relu`, `tanh` or
+        `sigmoid`. Set to `None` for no activation.
+    :param l2: Apply L2 normalization at the output layer.
+    """
+
+    name = 'mlp'
+    description = 'Simple MLP encoder trained from scratch'
+    task = 'any'
+    output_dim = '-'
+    architecture = 'MLP'
+
+    def __init__(
+        self,
+        input_size: int,
+        hidden_sizes: List[int],
```

Comment:
Sorry, I rethink it a little bit, and maybe this will be better? 
```
hidden_sizes: Optional[List[int]] = None,
output_size: int,
```
Users can use only one layer / multi-layers of MLP.
`output_size` is demanded since it's the output. However, `hidden_sizes` will only be used in multi-layers cases, so it's `Optional`.

For short, `hidden_layer` will not be directly connected to input or output, so it's hidden by the model.

Reply:
We removed output_size since it can be mixed with output_dim that is a tailor arg